cmd/kubectl-gadget/sigsnoop.go

```diff
@@ -0,0 +1,226 @@
+// +build !windows
+
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2019-2022 The Inspektor Gadget authors
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/sigsnoop/types"
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+	"github.com/spf13/cobra"
+
+	"golang.org/x/sys/unix"
+)
+
+var (
+	pid    int
+	sig    string
+	failed bool
+)
+
+func signalStringToIntString(signal string) (string, error) {
+	// There are three possibilities:
+	// 1. Either user did not give a signal, thus the argument is empty string.
+	// 2. Or signal begins with SIG.
+	// 3. Or signal is a string which contains an integer.
+	if signal == "" {
+		return "0", nil
+	}
+
+	if strings.HasPrefix(signal, "SIG") {
+		signalNum := unix.SignalNum(signal)
+		if signalNum == 0 {
+			return "", fmt.Errorf("no signal found for %q", signal)
+		}
+
+		return strconv.Itoa(int(signalNum)), nil
+	}
+
+	_, err := strconv.ParseInt(signal, 10, 32)
+	if err != nil {
+		return "", err
+	}
+
+	return signal, nil
+}
+
+func signalIntToString(signal int) string {
+	return unix.SignalName(syscall.Signal(signal))
+}
+
+var sigsnoopCmd = &cobra.Command{
+	Use:   "sigsnoop",
+	Short: "Trace signals issued by syscalls system-wide",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var transform func(string) string
+
+		signalString, err := signalStringToIntString(sig)
+		if err != nil {
+			return err
+		}
+
+		switch params.OutputMode {
+		case utils.OutputModeJson: // don't print any header
+		case utils.OutputModeCustomColumns:
+			fmt.Println(getCustomSigsnoopColsHeader(params.CustomColumns))
+			transform = formatEventSigsnoopCustomCols
+		case utils.OutputModeColumns:
+			transform = sigsnoopTransformLine
+
+			fmt.Printf("%-16s %-16s %-16s %-16s %-6s %-16s %-9s %-6s %-6s\n",
+				"NODE", "NAMESPACE", "POD", "CONTAINER",
+				"PID", "COMM", "SIGNAL", "TPID", "RET")
+		}
+
+		fmt.Printf("signal: %q\n", signalString)
+		fmt.Printf("pid: %q\n", strconv.Itoa(pid))
+
+		config := &utils.TraceConfig{
+			GadgetName:       "sigsnoop",
+			Operation:        "start",
+			TraceOutputMode:  "Stream",
+			TraceOutputState: "Started",
+			CommonFlags:      &params,
+			Parameters: map[string]string{
+				"signal": signalString,
+				"pid":    strconv.Itoa(pid),
+				"failed": strconv.FormatBool(failed),
+			},
+		}
+
+		return utils.RunTraceAndPrintStream(config, transform)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(sigsnoopCmd)
+	utils.AddCommonFlags(sigsnoopCmd, &params)
+
+	sigsnoopCmd.PersistentFlags().IntVarP(
+		&pid,
+		"pid",
+		"",
+		0,
+		"Show only signal sent by this particular PID",
+	)
+	sigsnoopCmd.PersistentFlags().StringVarP(
+		&sig,
+		"signal",
+		"",
+		"0",
+		`Show only events where this particular signal is sent (signal can be an int like 9 or string beginning by "SIG" like "SIGKILL")`,
```

Comment:
I think it's too long, what about something like:

`Trace only this signal (it can be an int like 9 or string beginning by "SIG" like "SIGKILL")`,